crates/rooch-framework/sources/address_mapping.move

```diff
@@ -21,19 +22,29 @@ module rooch_framework::address_mapping{
         mapping: Table<MultiChainAddress, address>,
     }
 
-    public(friend) fun genesis_init(ctx: &mut Context, genesis_account: &signer) {
+    public(friend) fun genesis_init(ctx: &mut Context, _genesis_account: &signer) {
         let mapping = context::new_table<MultiChainAddress, address>(ctx);
-        context::move_resource_to(ctx, genesis_account, AddressMapping{
+        context::new_singleton(ctx, AddressMapping{
```

Comment:
What are the advantages of `singleton` mode compared to `resource` mode?

Reply:
Singleton Object can be a transaction argument, so the developer can directly pass `Object<AddressMapping>` as a transaction argument without `Context`.